pulser-core/pulser/sequence/sequence.py

```diff
@@ -374,6 +375,132 @@ def config_slm_mask(self, qubits: Iterable[QubitId]) -> None:
         # If checks have passed, set the SLM mask targets
         self._slm_mask_targets = targets
 
+    def switch_device(
+        self, new_device: Device, strict: bool = False
+    ) -> Sequence:
+        """Switch the device of a sequence.
+
+        Args:
+            new_device: The target device instance.
+            strict: Enforce a strict match between devices and channels to
+                guarantee the pulse sequence is left unchanged.
+
+        Returns:
+            The sequence on the new device, using the matching channels of
+            the former device declared in the sequence.
+        """
+        # Check if the device is new or not
+
+        if self._device == new_device:
+            warnings.warn(
+                "Switching a sequence to the same device"
+                + " returns the sequence unchanged.",
+                stacklevel=2,
+            )
+            return self
+        if new_device == MockDevice:
+            raise NotImplementedError(
+                "Switching the device of a sequence"
+                + " to 'MockDevice' is not supported."
+            )
+        # Initialize the new sequence
+        new_seq = Sequence(self.register, new_device)
+
+        # Channel Matching
+
+        matched = False
+        strict_error_message = None
+        ch_type_er_mess = None
+        channel_matching = {}
+        sample_seq = sample(self)
+        alert_phase_jump = False
```

Comment:
This channel matching section will need to be updated and potentially moved to a method inside the device classes.

Reply:
I will think about it and propose.